client/allocrunner/alloc_runner.go

```diff
@@ -209,7 +209,8 @@ type RPCer interface {
 }
 
 // NewAllocRunner returns a new allocation runner.
-func NewAllocRunner(config *Config) (*allocRunner, error) {
+func NewAllocRunner(cfg any) (interfaces.AllocRunner, error) {
+	config := cfg.(*Config)
```

Comment:
How awful would it be to move allocrunner.Config somewhere like client/structs? :grimacing:

Reply:
Kinda rough. But maybe I can find a spot for it still.